src/core/thread/mesh_forwarder.cpp

```diff
@@ -515,7 +515,7 @@ ThreadError MeshForwarder::UpdateIp6Route(Message &aMessage)
             if (mMle.IsRoutingLocator(ip6Header.GetDestination()))
             {
                 rloc16 = HostSwap16(ip6Header.GetDestination().mFields.m16[7]);
-                VerifyOrExit(mMle.GetRouterId(rloc16) < Mle::kMaxRouterId, error = kThreadError_Drop);
+                VerifyOrExit(mMle.IsRouterIdCorrect(mMle.GetRouterId(rloc16)), error = kThreadError_Drop);
```

Comment:
Well, reference to a static function through object seems to be used quite often in OpenThread, that's why I did it this way. Moreover, a reference to `Mle::` namespace would also be needed in this case so the function call will be even longer. I also opt to leave it as it is.